block.go

```diff
@@ -0,0 +1,53 @@
+package sdk
+
+import (
+	"context"
+
+	ctypes "github.com/cometbft/cometbft/rpc/core/types"
+	cosmos "github.com/cosmos/cosmos-sdk/client"
+)
+
+// NodeStatusFetcher returns the status of the node to which it is connected.
+//
+// This status is used here in the BlockClient in order to get the latest block height.
+type NodeStatusFetcher interface {
+	Status(ctx context.Context) (*ctypes.ResultStatus, error)
+}
+
+// BlockClient is the interface to interact with the on-chain block module.
+//
+// For example, it can be used to get the latest block height.
+//
+// It uses the StatusClient interface of the cometbft to pull the latest block height.
+// This is done to avoid fetching the entire latest block just to extract the block height.
+//
+// The implementations of this interface could leverage websockets to get notified
+// about new blocks and cache the latest block height to avoid querying the blockchain
+// for it every time.
+type BlockClient struct {
+	NodeStatusFetcher
+}
+
+// NewBlockClient creates a new block client with the provided RPC URL.
+func NewBlockClient(queryNodeRPCUrl string) (*BlockClient, error) {
+	// TODO_IMPROVE: drop the cosmos dependency and directly use cometbft rpchttp.New, once the latter publishes a release that includes this functionality.
+	// Directly using the cometbft will simplify the code by both reducing imported repos and removing the cosmos wrapper which we don't use.
+	statusFetcher, err := cosmos.NewClientFromNode(queryNodeRPCUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	return &BlockClient{
+		NodeStatusFetcher: statusFetcher,
+	}, nil
+}
+
+// LatestBlockHeight returns the height of the latest committed block in the blockchain.
```

Comment:
Could/should we add this to the interface?

As a consumer, I don't know what's available on the structures I'm importing.

Reply:
- To keep interfaces on the consumer side, we need to return structs.
- I have updated the comments on the struct and its method, and added a `block_test.go` with an example of the usage. These are `go doc` friendly, which I think we can leverage to generate go doc references for the SDK.